src/components/MainPage.js

```diff
@@ -176,31 +240,29 @@ class MainPage extends Component<Props, State> {
                   </Grid>
                 ))}
               </Grid>
+              <Grid>
+                <Pagination
+                  prevPageText="prev"
+                  nextPageText="next"
+                  firstPageText="first"
+                  lastPageText="last"
+                  activePage={this.state.activePage}
+                  itemsCountPerPage={10}
+                  totalItemsCount={total_pages}
+                  pageRangeDisplayed={5}
+                  onChange={this.handlePageChange}
+                />
+              </Grid>
             </Grid>
           </Grid>
-          <Grid>
-            <BottomNavigation
-              value="recents"
-              onChange={this.handleBottomNavigationChange}
-              className={classes.bottomNav}
-            >
-              <BottomNavigationAction label="back" icon={<ArrowBack />} />
-              <BottomNavigationAction
-                label="Favorites"
-                onClick={this.openListOfFavorites}
-                icon={<FavoriteIcon />}
-              />
-              <BottomNavigationAction label="forward" icon={<ArrowForward />} />
-            </BottomNavigation>
-          </Grid>
         </main>
       );
     }
   }
 }
 
 function mapStateToProps(state) {
-  return { data: state.movies, isLoaded: state.isLoaded };
+  return { fetchedDataBundle: state.movies, isLoaded: state.isLoaded };
 }
 
 const mapDispatchToProps = (dispatch: Dispatch<>) => {

```

Comment:
I just saw - `Dispatch<>` is a very weird type :) I think must be `Dispatch<Action>`.

Or add flow-types for `react-redux` and a `redux`

Reply:
Done